### PR TITLE
feat: UX-friendly `lotus-miner sectors extend` cmd

### PR DESCRIPTION
**Pull request by beck-8**
## Related Issues

Closes: https://github.com/filecoin-project/lotus/issues/10967

## Proposed Changes

- [x] Simulates how much the message will cost
- [x] Adds that the --really-do-it flag needs to be added to actually send the message

Example:

```
lotus-miner sectors extend --from=786831 --to=786832
Extending 5 sectors: 
 {
  "Extensions": [
    {
      "Deadline": 5,
      "Partition": 0,
      "Sectors": "5,8",
      "NewExpiration": 1570600
    },
    {
      "Deadline": 2,
      "Partition": 0,
      "Sectors": "6",
      "NewExpiration": 1570600
    },
    {
      "Deadline": 3,
      "Partition": 0,
      "Sectors": "7,10",
      "NewExpiration": 1570600
    }
  ]
}
Simulating extending 5 sector(s) - estimated costs 0.000008576090848024 FIL
You need to pass the --really-do-it flag to actually send the message
```

With the --really-do-it flag:
```
lotus-miner sectors extend --really-do-it --from=786831 --to=786832
Extending 9 sectors: bafy2bzacedputunig7mqmldbfwm2dr3xahc6m26ln7sm5e3hd5jnm7qu4aovc
Simulating extending 9 sector(s) - estimated costs 0 FIL
```

## Additional Info

Manual testing covered (tested by phi-rjan)

- [x] Extend sectors with the `--from` and `--to` flag
- [x] Extend a sector by just adding the sector number after `lotus-miner sectors extend`
- [x] Extend sectors with the `--sectorfile` option

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: cmd/lotus-miner/sectors.go [hunk 1]
```diff
@@ -1389,6 +1390,25 @@ var sectorsExtendCmd = &cli.Command{
...
+
+				estMsg, err := fullApi.GasEstimateMessageGas(ctx, &types.Message{
+					From:   mi.Worker,
+					To:     maddr,
+					Method: builtin.MethodsMiner.ExtendSectorExpiration2,
+					Value:  big.Zero(),
+					Params: sp,
+				}, nil, types.EmptyTSK)
+				if err != nil && xerrors.Is(err, &api.ErrOutOfGas{}) {
+					return xerrors.Errorf("out of gas, please use --max-sectors to reduce the number of single message sectors")
```
Comment (rjan90) [inline cmd/lotus-miner/sectors.go:1407]:
```suggestion
					return xerrors.Errorf("out of gas, please use --max-sectors and reduce the amount of sectors you extend in each message")
```

File: cmd/lotus-miner/sectors.go [hunk 2]
```diff
@@ -1411,7 +1431,10 @@ var sectorsExtendCmd = &cli.Command{
 			fmt.Println(smsg.Cid())
 		}
 
-		fmt.Printf("%d sectors extended\n", stotal)
+		fmt.Printf("Simulating extending %d sector - estimated costs %s\n", stotal, types.FIL(gasTotal))
+		if !cctx.Bool("really-do-it") {
+			fmt.Println("You need to pass the --really-do-it flag to actully send the message")
```
Comment (rjan90) [inline cmd/lotus-miner/sectors.go:1436]:
```suggestion
		fmt.Printf("Simulating extending %d sector(s) - estimated costs %s\n", stotal, types.FIL(gasTotal))
		if !cctx.Bool("really-do-it") {
			fmt.Println("You need to pass the --really-do-it flag to actually send the message")
```

**Comment by rjan90** [comment]
Thank you so much for the PR! 💯 I added some small suggestions, I will leave the review of the code to the devs!

**Comment by rjan90** [comment]
I updated your orignal posting expanding on how the command now looks, and also some of the test that I covered during my initial testing here.

**Comment by beck-8** [comment]
I renewed a 64G node, with a total of 96526 sectors, 5.8PiB of data, and a maximum of 5000 sectors for each message. The final estimated result is that it will cost 1.4699FIL. This theoretical data does not meet expectations. Renewal then It is impossible to spend a little gas on multiple sectors, so I printed the GasFeeCap and GasLimit of each message in this place, and found that the GasFeeCap is very small? I don't know why this is happening.

>https://github.com/beck-8/lotus/blob/b4bbd09b59d359cafd233b9ddf4e15fc2b483ff0/cmd/lotus-miner/sectors.go#L1410
>fmt.Println(estMsg.GasFeeCap, estMsg.GasLimit, estMsg.RequiredFunds())
```bash
./lotus-miner sectors extend --to 9974920 --max-sectors 5000
...
Extending 4600 sectors: 
...
17038332 4108383150 69999996092905800
Simulating extending 96526 sector(s) - estimated costs 1.469999954001626457 FIL
You need to pass the --really-do-it flag to actually send the message
```

**Pull request closed without merging**